### PR TITLE
[ENH] Apply arbitrary command to some/all pyAFQ outputs, more BIDSy names

### PR DESCRIPTION
**Pull request by 36000**
This PR has two parts, both affect output file names.
1. Propose a framework for easily handling pyAFQ outputs according to what they dependent. In this case, we use a depends field we put in the json files. Also, you can exclude particular outputs if you want. This is, I think, a more elegant solution to #799 and #795
2. Make pyAFQ outputs follow a more BIDS-y format. Discussed previously in #819

### CODE REVIEW

**Comment by 36000** [comment]
@arokem  this is ready for review. I moved the dependent information to the json file.

File: AFQ/tasks/segmentation.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def segment(dwi, data_imap, mapping_imap,
 
 
 @pimms.calc("clean_bundles")
-@as_file('-clean_tractography.trk', include_track=True, include_seg=True)
+@as_file('-cleanTractography.trk', include_track=True, include_seg=True)
```
Comment (arokem) [inline AFQ/tasks/segmentation.py:86]: `_tractography` will eventually be a recognized BIDS suffix, so I think this should be OK without the change. Same is true for `_xform`.

**Comment by arokem** [comment]
I think that we should adhere to suffixes that are already or will eventually be part of the BIDS standard: `_mask`, `_xform`, `_tractography`. The problem is then what to do with `clean`. I think we can make that part of the `desc` portion of the filename, which can maybe be camel-cased. WDYT?

**Comment by 36000** [comment]
Putting in the desc makes sense to me!

**Comment by 36000** [comment]
Can now address #795  as well.

**Comment by 36000** [comment]
@arokem  I updated the top comment so this is ready to review / merge now.

**Comment by arokem** [comment]
We should follow up at some point add some documentation explaining how to "trim" and rerun pyAFQ while changing parameters (I think that's probably a typical use-case).

**Pull request merged**